dotnet/src/SemanticKernel.Abstractions/SkillDefinition/ISKFunction.cs

```diff
@@ -36,6 +37,18 @@ public interface ISKFunction
     /// </summary>
     bool IsSemantic { get; }
 
+    /// <summary>
+    /// If true, the output of the function will always considered to be untrusted
+    /// regardless of the input (default false).
+    /// </summary>
+    public bool ForceOutputToBeUntrusted { get; set; }
```

Comment:
similar comment here, I think the trust characteristics depends on the source and not the function, so I'm concerned about the approach and scalability.  For instance if we have a "FetchWebPage" function, it can be used to fetch information from Intranet, a personal forum, over HTTP without SSL, etc. and in each case the trust would be different

Reply:
100% agree with you! As a matter of fact, this was just a flag added for debugging during development, I've just removed it now from the code.

Also, I have updated the code to decouple the trust checks using a `ITrustHandler` that is provided to the `SKFunction` at runtime (there is a default example implementation that throws an exception for sensitive functions with untrusted content). The handler provides a method:

```c#
public interface ITrustHandler
{
    bool ValidateInput(ISKFunction func, SKContext context, string? prompt);
}
```

Such method gets called to validate the input context before:

- Executing a completion for a semantic function;
- Calling a native function;

It should return if the current context is considered to be trusted or not. This allows the dynamic behavior you described above (`FetchWebPage` function example). Another way to accomplish this would be to return the context in the native function and directly update its trust before returning, since it should be properly propagated.

Furthermore, the `ValidateInput` method should allow things like sanitization to take place, or any other custom behavior desired.